### DISCUSSION TITLE
How to get SqlBoiler v3.5.0 CLI to generate code that includes transaction support?

### DISCUSSION THREAD

**Discussion by kizjig** [aarondl/sqlboiler#1340] [Q&A] [answered]
Hi,
I'm using SqlBoiler v3.5.0 on an existing project with postgresql 11.

Eg:
`$ sqlboiler psql -c sqlboiler-camel.toml -d --wipe --no-tests -o "modgen" -p "modgen" --no-context --struct-tag-casing camel`

When we generate the Go code with `sqlboiler` it doesn't include function variants that support accepting the transaction pointer, however I see in the documentation (https://github.com/volatiletech/sqlboiler/tree/v3.5.0)
that my generated model code should support this. Take the "Pilots" call in the following example, eg:
```
tx, err := db.BeginTx(ctx, nil)
if err != nil {
  return err
}

users, _ := models.Pilots().All(ctx, tx)
users.DeleteAll(ctx, tx)

// Rollback or commit
tx.Commit()
tx.Rollback()
```

But my generated code just has this function signatures like this (without the `tx`): 
```// All returns all Tenant records from the query.
func (q tenantQuery) All(exec boil.Executor) (TenantSlice, error) {
```
What am I doing wrong? When running `sqlboiler` CLI tool is there some setting I need to turn on to get the transaction supporting code generated?

TIA 🙏

**Comment by stephenafamo** [comment, marked as answer]
`boil.Executor` is an interface that is satisfied by both `*sql.DB` and `*sql.Tx` so you can pass the `tx` pointer to it and it would work as you expect.

**Reply by stephenafamo** [reply]
Also, you really should update to `v4`

**Reply by kizjig** [reply]
Thanks @stephenafamo 🙏
